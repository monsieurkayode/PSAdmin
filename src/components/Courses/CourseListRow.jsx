import React from 'react';
import { Link } from 'react-router-dom';
import { string, func } from 'prop-types';

/**
 *
 * @function CourseListRow
 *
 * @param {object} { id, watchHref, title, authorId, category, length }
 *
 * @returns {JSX}
 */
const CourseListRow = ({
  id, watchHref, title, authorId, category, length, deleteCourse
}) => (
  <tr>
    <td>
      <a href={watchHref} target="_blank" rel="noopener noreferrer">
        Watch
      </a>
    </td>
    <td><Link to={`/course/${id}`}>{title}</Link></td>
    <td>{authorId}</td>
    <td>{category}</td>
    <td>{length}</td>
    <td>
      <button
        type="button"
        className="btn btn-sm btn-danger"
        onClick={() => deleteCourse(id)}
      >
        Delete
      </button>
    </td>
  </tr>
);

CourseListRow.propTypes = {
  authorId: string.isRequired,
  category: string.isRequired,
  id: string.isRequired,
  length: string.isRequired,
  title: string.isRequired,
  watchHref: string.isRequired,
  deleteCourse: func.isRequired
};

export default CourseListRow;
