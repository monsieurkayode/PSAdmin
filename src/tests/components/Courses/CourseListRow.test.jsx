import React from 'react';
import { shallow } from 'enzyme';
import CourseListRow from '../../../components/Courses/CourseListRow';
import mockCourse from '../../__mocks__/mockCourse';

const setup = () => {
  const props = {
    ...mockCourse,
    deleteCourse: jest.fn()
  };

  const wrapper = shallow(<CourseListRow {...props} />);

  return {
    props,
    wrapper
  };
};

describe('CourseListRow Component', () => {
  const { wrapper, props } = setup();

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a table row six table data cells', () => {
    expect(wrapper.find('tr').length).toBe(1);
    expect(wrapper.find('td').length).toBe(6);
  });

  it('renders an anchor tag element with Watch text content', () => {
    expect(wrapper.find('a').length).toBe(1);
    expect(wrapper.find('a').text()).toBe('Watch');
  });

  it('renders a Link component', () => {
    expect(wrapper.find('Link').length).toBe(1);
  });

  it('renders a Delete button', () => {
    expect(wrapper.find('button').length).toBe(1);
    expect(wrapper.find('button').text()).toBe('Delete');
  });

  it('calls deleteCourse onClick of Delete button', () => {
    const deleteButton = wrapper.find('button');
    deleteButton.simulate('click');

    expect(props.deleteCourse).toHaveBeenCalled();
    expect(props.deleteCourse).toHaveBeenLastCalledWith(mockCourse.id);
  });
});
