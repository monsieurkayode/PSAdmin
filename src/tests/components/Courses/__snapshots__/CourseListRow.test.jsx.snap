// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseListRow Component renders correctly 1`] = `
<tr>
  <td>
    <a
      href="react-flux-building-applications"
      rel="noopener noreferrer"
      target="_blank"
    >
      Watch
    </a>
  </td>
  <td>
    <Link
      replace={false}
      to="/course/react-flux-building-applications"
    >
      Building Applications in React and Flux
    </Link>
  </td>
  <td>
    cory-house
  </td>
  <td>
    JavaScript
  </td>
  <td>
    5:08
  </td>
  <td>
    <button
      className="btn btn-sm btn-danger"
      onClick={[Function]}
      type="button"
    >
      Delete
    </button>
  </td>
</tr>
`;
